sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -206,18 +205,16 @@ public static void main(
 
       LoadingCache<String, BeamFnApi.ProcessBundleDescriptor> processBundleDescriptors =
           CacheBuilder.newBuilder()
+              .maximumSize(1000)
+              .expireAfterAccess(10, TimeUnit.MINUTES)
               .build(
                   new CacheLoader<String, BeamFnApi.ProcessBundleDescriptor>() {
                     @Override
                     public BeamFnApi.ProcessBundleDescriptor load(String id) {
-                      try {
-                        return blockingControlStub.getProcessBundleDescriptor(
-                            BeamFnApi.GetProcessBundleDescriptorRequest.newBuilder()
-                                .setProcessBundleDescriptorId(id)
-                                .build());
-                      } catch (Throwable th) {
-                        return (BeamFnApi.ProcessBundleDescriptor) fnApiRegistry.getById(id);
```

Comment:
do we not need the try/catch block anymore?

Reply:
All runners now support the pull protocol.